internal/dag/gatewayapi_processor.go

```diff
@@ -1417,7 +1426,7 @@ func gatewayGRPCHeaderMatchConditions(matches []gatewayapi_v1alpha2.GRPCHeaderMa
 		seenNames.Insert(upperName)
 
 		headerMatchConditions = append(headerMatchConditions, HeaderMatchCondition{
-			MatchType: HeaderMatchTypeExact,
+			MatchType: matchType,
```

Comment:
For regex, we should validate that it's a valid regex and return an error if not. We have a `ValidateRegex` function in this package that you can use.